### PR TITLE
Enable lint check for comments

### PR DESCRIPTION
**Pull request by siggy**
Part of #217. Follow up from #1982 and #2018.

A subsequent commit will fix the ci failure.

Signed-off-by: Andrew Seigner <siggy@buoyant.io>

### 2nd commit

Address all comment-related linter errors.

This change addresses all comment-related linter errors by doing the
following:
- Add comments to exported symbols
- Make some exported symbols private
- Recommend via TODOs that some exported symbols should should move or
  be removed

This PR does not:
- Modify, move, or remove any code
- Modify existing comments

Signed-off-by: Andrew Seigner <siggy@buoyant.io>

### CODE REVIEW

**Comment by siggy** [comment]
Relates to #216.

File: controller/api/proxy/client.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"google.golang.org/grpc"
 )
 
+// NewClient creates a new gRPC client to the Destination service.
+// TODO: consider moving this into destination-client, or removing altogether.
```
Comment (klingerf) [inline controller/api/proxy/client.go]: 👍 Looks like this is only called from the destination-client script, so I'm definitely in favor of removing this file and moving the code directly into the script. But doesn't necessarily have to happen as part of this branch.

File: controller/api/public/client.go [hunk 2]
```diff
@@ -190,6 +190,7 @@ func newClient(apiURL *url.URL, httpClientToUse *http.Client, controlPlaneNamesp
 	}, nil
 }
 
+// NewInternalClient creates a new Public API client intended to run inside a Kubernetes cluster.
```
Comment (klingerf) [inline controller/api/public/client.go]: Mind wrapping this at 80 characters? Same goes for the comment on line 203 in this file.

File: controller/api/public/test_helper.go [hunk 3]
```diff
@@ -110,37 +121,41 @@ func (a *MockAPITapByResourceClient) Recv() (*pb.TapEvent, error) {
...
 //
 
-type MockProm struct {
+type mockProm struct {
 	Res             model.Value
 	QueriesExecuted []string // expose the queries our Mock Prometheus receives, to test query generation
 	rwLock          sync.Mutex
 }
 
+// PodCounts is a test helper struct that for representing data in a
```
Comment (klingerf) [inline controller/api/public/test_helper.go]: Typo here. Maybe it should be: // PodCounts is a test helper struct that _is used_ for representing data in a

File: controller/api/util/api_utils.go [hunk 4]
```diff
@@ -535,6 +547,8 @@ func routeLabels(event *pb.TapEvent) string {
 	return out
 }
 
+// RenderTapEvent renders a Public API TapEvent to a string.
+// TODO: consider moving this into cli/cmd/tap.go.
```
Comment (klingerf) [inline controller/api/util/api_utils.go]: 👍 Ah, right, this was originally moved here when we were rendering tap output as plaintext in the web UI, but we could move it back now that the web has its own rendering.

File: controller/k8s/api.go [hunk 5]
```diff
@@ -140,62 +143,71 @@ func (api *API) Sync() {
...
 
+// Pod provides access to a shared informer and lister for Pods.
 func (api *API) Pod() coreinformers.PodInformer {
 	if api.pod == nil {
 		panic("Pod informer not configured")
 	}
 	return api.pod
 }
 
+// RC provides access to a shared informer and lister for ReplicaSets.
```
Comment (klingerf) [inline controller/k8s/api.go]: Typo here: `ReplicaSets` => `ReplicationControllers`

File: controller/api/proxy/client.go (hunk 1 above)
Comment (siggy) [reply controller/api/proxy/client.go]: Sounds good, will address in a subsequent PR.

File: controller/api/util/api_utils.go (hunk 4 above)
Comment (siggy) [reply controller/api/util/api_utils.go]: Will do in a subsequent PR.

**Pull request merged**